Mathlib/Data/Set/Lattice.lean

```diff
@@ -543,6 +543,14 @@ theorem directed_on_iUnion {r} {f : ι → Set α} (hd : Directed (· ⊆ ·) f)
     ⟨x, ⟨z, xf⟩, xa₁, xa₂⟩
 #align set.directed_on_Union Set.directed_on_iUnion
 
+theorem directed_on_sUnion {r} {S : Set (Set α)} (hd : DirectedOn (· ⊆ ·) S)
+    (h : ∀ x ∈ S, DirectedOn r x) : DirectedOn r (⋃₀ S) := by
+  simp only [DirectedOn, mem_sUnion, exists_imp]
```

Comment:
Can this proof reuse the theorem above?

Reply:
I would suggest doing the rename suggested above, and moving both lemmas to below `sUnion_eq_iUnion` and proving the latter as

```lean
theorem directedOn_sUnion {r} {S : Set (Set α)} (hd : DirectedOn (· ⊆ ·) S)
    (h : ∀ x ∈ S, DirectedOn r x) : DirectedOn r (⋃₀ S) := by
  rw [sUnion_eq_iUnion]
  exact directed_on_iUnion (directedOn_iff_directed.mp hd) (fun i ↦ h i.1 i.2)
```

Reply:
Thanks! I did what Dagur suggested.